src/internal/functions/Remove-AzOpsDeployment.ps1

```diff
@@ -0,0 +1,164 @@
+            $roleAssignments = Get-AzRoleAssignment -ServicePrincipalName $context.Account.id -Scope $policyAssignment.Properties.Scope -ErrorAction Continue
+            foreach($role in $roleAssignments){
+                if ( $role.RoleDefinitionName -eq "Owner" -or $role.RoleDefinitionName -eq "User Access Administrator" -or
+                (Get-AzRoleDefinition -Name $role.RoleDefinitionName | Where-Object {$_.Actions -contains "Microsoft.Authorization/policyAssignments/delete" -or $_.Actions -contains "Microsoft.Authorization/policyAssignments/*" -or $_.Actions -contains "Microsoft.Authorization/*"})){
+                    $roleAssignmentPermissionCheck = $true
+                }
+            }
+            if (($resultsError -ne $null) -or (-not $policyAssignment)) {
+                Write-PSFMessage -Level Warning -String 'Remove-AzOpsDeployment.RemovePolicyAssignment.NoPolicyAssignmentFound' -StringValues $scopeObject.Scope,$resultsError -Target $scopeObject
+                $results = '{0}: What if Operation Failed: Performing the operation "Deleting the policy assignment..." on target {1}.' -f $deploymentName, $scopeObject.scope
+                Set-AzOpsWhatIfOutput -results $results -removeAzOpsFlag $true
+                return
+            }
+            elseif ((-not $roleAssignmentPermissionCheck)) {
+                Write-PSFMessage -Level Warning -String 'Remove-AzOpsDeployment.RemoveAssignment.MissingPermissionOnContext' -StringValues $context.Account.Id,$scopeObject.Scope -Target $scopeObject
+                $results = '{0}: What if Operation Failed: Performing the operation "Deleting the policy assignment..." on target {1}.' -f $deploymentName,$scopeObject.scope
+                Set-AzOpsWhatIfOutput -results $results -removeAzOpsFlag $true
+                return
+            }
+            else {
+                $results = '{0}: What if Successful: Performing the operation "Deleting the policy assignment..." on target {1}.' -f $deploymentName,$scopeObject.scope
+                Write-PSFMessage -Level Verbose -String 'Set-AzOpsWhatIfOutput.WhatIfResults' -StringValues $results -Target $scopeObject
+                Write-PSFMessage -Level Verbose -String 'Set-AzOpsWhatIfOutput.WhatIfFile' -Target $scopeObject
+                Set-AzOpsWhatIfOutput -results $results -removeAzOpsFlag $true
+            }
+
+            #removal of resource
+            if ($PSCmdlet.ShouldProcess("RemovePolicyAssignment?")) 
+            {
+                Remove-AzPolicyAssignment -Id $scopeObject.scope -ErrorAction Stop
+            }
+            else{
+                Write-PSFMessage -Level Verbose -String 'Remove-AzOpsDeployment.SkipDueToWhatIf'
+            }
+        }
+        #endregion PolicyAssignment
+
+        #Region roleAssignments
+        if($scopeObject.Resource -eq "roleAssignments")
+        {
+            #Validate
+            $roleAssignmentPermissionCheck = $false
```

Comment:
To avoid duplicating code, can we create a generic function outside of this function to validate the permissions?

Reply:
Agree, Generic function has been added.